clang/lib/Basic/CodeGenOptions.cpp

```diff
@@ -11,46 +11,13 @@
 
 namespace clang {
 
-CodeGenOptions::CodeGenOptions() {
-#define CODEGENOPT(Name, Bits, Default) Name = Default;
-#define ENUM_CODEGENOPT(Name, Type, Bits, Default) set##Name(Default);
-#include "clang/Basic/CodeGenOptions.def"
+CodeGenOptions::CodeGenOptions() { resetNonModularOptions(); }
```

Comment:
I find this confusing: why are we resetting only the non-modular options here? I'd expect all options to be set to default here. Maybe create extra helper function `resetAllOptions()` that will call to `resetNonModularOptions()` and explain how modular options are handled?

Reply:
Right now the code is the same. I modified it and added a new method to reset all options with a fixme to actually reset all options. I prefer to do that in a separate commit, because fixing that is a different change from what I am trying to accomplish in this PR.